fhir-search/src/main/java/com/ibm/fhir/search/util/ReferenceUtil.java

```diff
@@ -62,18 +62,68 @@ public static ReferenceValue createReferenceValueFrom(Reference ref, String base
         Integer version = null;
 
         if (ref.getReference() != null && ref.getReference().getValue() != null) {
+            return createReferenceValueFrom(ref.getReference().getValue(), ref.getType() != null ? ref.getType().getValue() : null, baseUrl);
+        } else if (ref.getIdentifier() != null && ref.getIdentifier().getValue() != null) {
+            // LOGICAL REFERENCE
+            value = ref.getIdentifier().getValue().getValue();
+            referenceType = ReferenceType.LOGICAL;
+            if (ref.getType() != null) {
+                // We need
+                targetResourceType = ref.getType().getValue();
```

Comment:
What happens when the type is null? similar to prior comment?

Reply:
In the case of an identifier, it is not required that the type is specified. The identifier is valid on it's own.